crud/insert.lua

```diff
@@ -63,30 +60,30 @@ function insert.call(space_name, obj, opts)
     end
     local space_format = space:format()
 
-    -- compute default bucket_id
-    local tuple, err = utils.flatten(obj, space_format)
-    if err ~= nil then
-        return nil, InsertError:new("Object is specified in bad format: %s", err)
-    end
-
     local key = utils.extract_key(tuple, space.index[0].parts)
-
     local bucket_id = vshard.router.bucket_id_strcrc32(key)
     local replicaset, err = vshard.router.route(bucket_id)
     if replicaset == nil then
         return nil, InsertError:new("Failed to get replicaset for bucket_id %s: %s", bucket_id, err.err)
     end
 
-    local tuple, err = utils.flatten(obj, space_format, bucket_id)
+    local bucket_id_fieldno, err = utils.get_bucket_id_field(space)
     if err ~= nil then
-        return nil, InsertError:new("Object is specified in bad format: %s", err)
+        return nil, err
     end
 
+    -- Usually it's nil but to avoid unexpected behaviour for users
+    -- let's save previous value of "bucket_id" field.
+    local prev_bucket_id_value = tuple[bucket_id_fieldno]
+    tuple[bucket_id_fieldno] = bucket_id
```

Comment:
I would like it to be

```
tuple = table.copy(tuple)
```

and return modified object.

This way we

* don't change user's tuple
* return exactly inserted tuple, like we should do.

Moreover, if user passed tuple with specified `bucket_id`, we simply insert tuple with right `bucket_id` and then return a tuple with a user-specified `bucket_id`. WTF?

Reply:
> I would like it to be

No, it makes suffer Lua GC. I don't want to sacrifice performance.

Reply:
Ok, but I see undefined behaviour here.

> Moreover, if user passed tuple with specified `bucket_id`, we simply insert tuple with right `bucket_id` and then return a tuple with a user-specified `bucket_id`. WTF?

Reply:
It's not UB

But well, let's return error in such cases.

Reply:
You mean return an error if user passed object/tuple with `bucket_id` specified? I agree.

Reply:
Changed to error